src/lean_spec/subspecs/forkchoice/store.py

```diff
@@ -164,210 +175,408 @@ def process_attestation(
+        #
+        # This handles the case where no states have updated justification.
+        # Maintains the previous justified checkpoint rather than losing it.
+        if latest_justified is None:
+            latest_justified = self.latest_justified
 
-    def accept_new_attestations(self) -> None:
+        # Run LMD-GHOST fork choice algorithm
+        #
+        # Selects canonical head by walking the tree from the justified root,
+        # choosing the heaviest child at each fork based on attestation weights.
+        new_head = get_fork_choice_head(
+            self.blocks,
+            latest_justified.root,
+            self.latest_known_attestations,
+        )
+
+        # Extract finalized checkpoint from head state
+        #
+        # The head state tracks the highest finalized checkpoint. If the
+        # head changed, we may have a new finalized checkpoint.
+        #
+        # Fallback to current finalized if head state unavailable (defensive).
+        latest_finalized = (
+            self.states[new_head].latest_finalized
+            if new_head in self.states
+            else self.latest_finalized
+        )
+
+        # Return new Store instance with updated values (immutable update)
+        return self.model_copy(
+            update={
+                "head": new_head,
+                "latest_justified": latest_justified,
+                "latest_finalized": latest_finalized,
+            }
+        )
+
+    def accept_new_attestations(self) -> "Store":
         """
         Process pending attestations and update forkchoice head.
 
-        Moves attestations from latest_new_attestations to latest_known_attestations and triggers
-        head update.
+        Moves attestations from latest_new_attestations to latest_known_attestations,
+        making them eligible to contribute to fork choice weights. This migration
+        happens at specific interval ticks.
+
+        The Interval Tick System
+        -------------------------
+        Attestations progress through intervals:
+        - Interval 0: Block proposal
+        - Interval 1: Validators cast attestations (enter "new")
+        - Interval 2: Safe target update
+        - Interval 3: Attestations accepted (move to "known")
+
+        This staged progression ensures proper timing and prevents premature
+        influence on fork choice decisions.
+
+        Returns:
+            New Store with migrated attestations and updated head.
         """
         # Move all new attestations to known attestations
-        for validator_id, signed_attestation in self.latest_new_attestations.items():
-            self.latest_known_attestations[validator_id] = signed_attestation
+        new_known = dict(self.latest_known_attestations)
+        new_known.update(self.latest_new_attestations)
+
+        # Create store with migrated attestations
+        store = self.model_copy(
+            update={
+                "latest_known_attestations": new_known,
```

Comment:
```suggestion
                "latest_known_attestations": {**self.latest_known_attestations, **self.latest_new_attestations},
```

Or we might use the merge operator for dictionary like:

```python
self.latest_known_attestations | self.latest_new_attestations
```

Reply:
thanks much better, fixed